pkg/server/auth/server.go

```diff
@@ -171,21 +175,36 @@ func claimsConfigFromSecret(secret corev1.Secret) *ClaimsConfig {
 	return nil
 }
 
-func NewAuthServerConfig(log logr.Logger, oidcCfg OIDCConfig, kubernetesClient ctrlclient.Client, tsv TokenSignerVerifier, namespace string, authMethods map[AuthMethod]bool, noAuthUser string) (AuthServerConfig, error) {
-	return AuthServerConfig{
+// NewAuthServerConfig creates and returns a new AuthServerConfig.
+//
+// The oidcCfg.IssuerURL and oidcCfg.RedirectURL are given a light validation to
+// ensure they are valid URLs.
+func NewAuthServerConfig(log logr.Logger, oidcCfg OIDCConfig, kubernetesClient ctrlclient.Client, tsv TokenSignerVerifier, namespace string, authMethods map[AuthMethod]bool, noAuthUser string, sm SessionManager) (*AuthServerConfig, error) {
+	if authMethods[OIDC] {
+		if _, err := url.Parse(oidcCfg.IssuerURL); err != nil {
+			return nil, fmt.Errorf("invalid issuer URL: %w", err)
+		}
+
+		if _, err := url.Parse(oidcCfg.RedirectURL); err != nil {
+			return nil, fmt.Errorf("invalid redirect URL: %w", err)
+		}
+	}
```

Comment:
These two checks are in initAuthServer. They moved over there in #3958 as part of the the light refactoring to the very long arg lists. The thought was we could remove this `NewAuthServerConfig` function to and just create the AuthServerConfig literal.. or rename initAuthServer to become NewAuthServerConfig etc instead.

I think I stopped at that point as there were quite a few `NewAuthServerConfig`s around that would cause even more conflicts to resolve etc given the other inflight PRs..